### PR TITLE
Aggregate and proof subscriber

### PR DESCRIPTION
**Pull request by terencechain**
Part of #4224 

This PR adds in the aggregate and proof subscriber

Thanks @rauljordan for helping out with the patching for this one

### CODE REVIEW

File: beacon-chain/p2p/gossip_topic_mappings.go [hunk 1]
```diff
@@ -15,6 +16,7 @@ var GossipTopicMappings = map[string]proto.Message{
 	"/eth2/voluntary_exit":     &pb.VoluntaryExit{},
 	"/eth2/proposer_slashing":  &pb.ProposerSlashing{},
 	"/eth2/attester_slashing":  &pb.AttesterSlashing{},
+	"/eth2/beacon_aggregate_and_proof": &pbp2p.AggregateAndProof{},
```
Comment (terencechain) [inline beacon-chain/p2p/gossip_topic_mappings.go]: This ideally should be a definition in `ethereumapis`, we'll need to update `ethereumapis` for the latest aggregate definitions 

Happy to do that first

Comment (prestonvanloon) [reply beacon-chain/p2p/gossip_topic_mappings.go]: I think that's a good idea

**Comment by prestonvanloon** [comment]
Fails e2e tests. Please take a look

File: beacon-chain/p2p/gossip_topic_mappings.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/p2p/gossip_topic_mappings.go]: gofmt?

Comment (terencechain) [reply beacon-chain/p2p/gossip_topic_mappings.go]: will get to this tomorrow

**Comment by prestonvanloon** [comment]
Regarding e2e, what PR started it? We agreed to skip it, not ignore it. I'd rather keep it green as long as possible, then maybe for 2 or 3 PRs we can skip it to allow easier review. This branch is going to be a mess to fix if we ignore e2e for too long.

**Comment by prestonvanloon** [comment]
Screenshot from 2019-12-11 12-46-05

Looks like attestation pool doesn't work at runtime. This could be a big issue if ignored.

**Comment by terencechain** [comment]
I think we can skip it with clear reasoninh (which is still yet to be provided AI: Terence)  or else maintain it PR by PR can get really tedious. Attestation pool redesign breaks run time until the end

**Comment by prestonvanloon** [comment]
It shouldn’t be tedious to keep the client working at runtime. I’ll defer
to you if you want to own fixing e2e rather than PR authors ensuring their
changes don’t break e2e.

**Pull request merged**